WalletWasabi/WabiSabi/Client/AliceClient.cs

```diff
@@ -46,15 +46,15 @@ public async Task RegisterInputAsync(CancellationToken cancellationToken)
 			Logger.LogInfo($"Round ({RoundId}), Alice ({AliceId}): Registered an input.");
 		}
 
-		public async Task ConfirmConnectionAsync(TimeSpan confirmInterval, CancellationToken cancellationToken)
+		public async Task ConfirmConnectionAsync(CancellationToken cancellationToken)
 		{
-			while (!await ConfirmConnectionAsync(cancellationToken).ConfigureAwait(false))
+			while (!await TryConfirmConnectionAsync(cancellationToken).ConfigureAwait(false))
```

Comment:
What if we are just waiting for `Phase.ConnectionConfirmation` and then try to ConfirmConnectionAsync. Fewer requests.

Reply:
I think that would be possible (and IIRC Arena is written to tolerate this in the way it sets timeouts for alices). Connection confirmation requests are expensive for both server and client (range proofs), so this could make a big difference.

However, if clients drop out and disappear the coordinator might still think they are registered, which would cause a low confirmation rate for registered inputs. This might be enough of a reason to to keep this polling behavior, and only cut the input registration phase short if the number of recently confirmed inputs exceeds the minimum required.

I think so long as the interval is sufficiently long (on the order of up to a few minutes) the performance hit should be pretty small, and the benefit of ensuring that just because an input registered like half an hour ago, we don't assume the client is still around to confirm it when it's time would be that there wouldn't be an apparent drop in the number of registered inputs, and larger average tx size.

Reply:
Do we have a timeout in Arena for Alices in InputRegistration?

Reply:
@molnard yes, there is a timeout. We have to confirm the connection periodically for two reasons:

* Alices that fail to confirm will be timed out (removed) and,
* We are getting extra zero credentials in each confirmation.

Reply:
That timeout is currently the same for all alices, specified by the round phase timeouts (twice, in fact):

https://github.com/zkSNACKs/WalletWasabi/blob/master/WalletWasabi/WabiSabi/Backend/PostRequests/InputRegistrationHandler.cs#L111

https://github.com/zkSNACKs/WalletWasabi/blob/master/WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs#L318

We should probably change that so it matches the behavior Lucas is describing.

Reply:
Anyway, Ive included the `ConnectionConfirmationTimeout` as a property in the `ConstructionState` and used `0.5 * ConnectionConfirmationTimeout` as connection confirmation interval. In this way the clients can adapt to changes in the coordinator configuration.